src/custom_nodes/common/buffersqueue.cpp

```diff
@@ -29,10 +29,13 @@ BuffersQueue::BuffersQueue(size_t singleBufferSize, int streamsLength) :
         inferRequests.push_back(memoryPool.get() + i * singleBufferSize);
     }
 }
+
 void* BuffersQueue::getBuffer() {
     // can be easily switched to async version if need arise
     auto idleId = getIdleStream();
     if (idleId.wait_for(std::chrono::nanoseconds(0)) == std::future_status::timeout) {
+        auto id = idleId.get();  // temporary solution related to future usage
+        returnBuffer(getInferRequest(id));
```

Comment:
`idleId.get()` will make the thread wait for future to become ready. In this case what is the point of returning nullptr if we can return the buffer instead?

Reply:
As far as I know future get method waits for the valid result, so would it be okay to just do
`auto idleId = getIdleStream();
return getInferRequest(idleId.get());`?
Case: 2 clients send requests and only one stream is allocated, the second client that wants to get buffer used by the first one. What I think would happen is the first client would finish its work with it and return the buffer. Then the second client could get the returned buffer for itself.

I tested that solution and I think it is working, however with it we would drop idea to allocate memory when all pre-allocated buffers are used(thats because we would always wait for pre-allocated buffer instead of throwing nullptr when none is available). Let me know what you think @atobiszei @dkalinowski

Reply:
The issue here is that BuffersQueue always give you future to idleId which means that you have to return it eventually.
We don't want to wait for preallocated buffers if none is available so we will dynamically allocate. However issue is that we owe id to the buffers queue and we have to release it ...

Handling this would complicate custom lib code and potentially block buffer for longer than necessary. Changing queue API should be fixed in separate commit to not make this change too big. Especially that we can work around this for now and just preallocate more buffers than necessary.

Reply:
I mean that calling `idleId.get()` internally calls `::wait` forever and waits till future becomes available. This causes the `getBuffer` to return nullptr ONLY after we receive new idle buffer. Why should we return nullptr if we just received idle buffer? Why not to return the buffer instead? @atobiszei

Reply:
With changed implementation you would not get idleId, so it's better to already develop code that will work with such assumption.